wagtail/admin/views/pages/listing.py

```diff
@@ -218,53 +218,58 @@ def get_queryset(self):
-                ).order_by("null_position", "latest_revision_created_at")
-            elif self.ordering == "-latest_revision_created_at":
-                # order by oldest revision first.
-                # Special case NULL entries - these should go at the end of the list.
-                pages = pages.annotate(
-                    null_position=Count("latest_revision_created_at")
-                ).order_by("-null_position", "-latest_revision_created_at")
-            else:
-                pages = pages.order_by(self.ordering)
-
         # We want specific page instances, but do not need streamfield values here
         pages = pages.defer_streamfields().specific()
 
-        # allow hooks defer_streamfieldsyset
-        for hook in hooks.get_hooks("construct_explorer_page_queryset"):
-            pages = hook(self.parent_page, pages, self.request)
-
         # Annotate queryset with various states to be used later for performance optimisations
         if getattr(settings, "WAGTAIL_WORKFLOW_ENABLED", True):
             pages = pages.prefetch_workflow_states()
 
         pages = pages.annotate_site_root_state().annotate_approved_schedule()
 
+        return pages
+
+    def order_queryset(self, queryset):
+        if self.is_searching and not self.is_explicitly_ordered:
+            # search backend will order by relevance in this case, so don't bother to
+            # apply an ordering on the queryset
+            return queryset
+
+        if self.ordering == "ord":
+            # preserve the native ordering from get_children()
+            pass
+        elif self.ordering == "latest_revision_created_at":
+            # order by oldest revision first.
+            # Special case NULL entries - these should go at the top of the list.
+            # Do this by annotating with Count('latest_revision_created_at'),
+            # which returns 0 for these
+            queryset = queryset.annotate(
+                null_position=Count("latest_revision_created_at")
+            ).order_by("null_position", "latest_revision_created_at")
+        elif self.ordering == "-latest_revision_created_at":
+            # order by oldest revision first.
+            # Special case NULL entries - these should go at the end of the list.
+            queryset = queryset.annotate(
+                null_position=Count("latest_revision_created_at")
+            ).order_by("-null_position", "-latest_revision_created_at")
+        else:
+            queryset = super().order_queryset(queryset)
+
+        return queryset
+
+    def search_queryset(self, queryset):
+        # allow hooks to modify queryset. This should happen as close as possible to the
+        # final queryset, but (for backward compatibility) needs to be passed an actual queryset
+        # rather than a search result object
+        for hook in hooks.get_hooks("construct_explorer_page_queryset"):
+            queryset = hook(self.parent_page, queryset, self.request)
```

Comment:
I also explored the idea of adding a `modify_queryset` in 8b86e3f08118fa8eaa53ed0a7fd7c14407ba43b0, but I'm not sure if that's something we want in the `BaseListingView`.